app/src/main/java/com/babylon/wallet/android/presentation/main/MainViewModel.kt

```diff
@@ -93,8 +95,8 @@ class MainViewModel @Inject constructor(
              * even if the profile has not yet been restored
              */
             when (profileState) {
-                is ProfileState.Restored -> p2PLinksRepository.observeP2PLinks()
-                else -> p2PLinksRepository.observeP2PLinks().drop(1)
+                is ProfileState.Loaded -> p2PLinksRepository.observeP2PLinks()
+                else -> p2PLinksRepository.observeP2PLinks().drop(1) // TODO why that?
```

Comment:
This fixed the issue with the connection to the ledger when creating the first account (when the profile doesn't exist yet). If I remember correctly what happened was that during onboarding, when connecting to the ledger we save the p2p link, connect to it from there and it was "conflicting" with the connection logic here and breaking the flow (the connection to ledger wasn't working anymore). I've tried to encompass this in the comment above.

Reply:
I see